skxray/speckle_visibility/speckle_visibility.py

```diff
@@ -0,0 +1,290 @@
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_int_labels : dict
+        average intensity of each ROI as a dictionary
+        {roi 1: average intensities, roi 2 : average intensities}
+
+    """
+    return {n+1 : mean_intensity(images_set[n],
+                                 labels) for n in range(len(images_set))}
+
+
+def mean_intensity(images, labels):
+    """
+    Mean intensities for ROIS' of the labeled array for set of images
+
+    Parameters
+    ----------
+    images : array
+        iterable of 2D arrays
+        dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_int : array
+        mean intensity of each ROI for the set of images as an array
+        shape (number of images in the set, number of labels)
+
+    """
+    if labels.shape != images[0].shape[0:]:
+        raise ValueError("Shape of the images should be equal to"
+                         " shape of the label array")
+
+    index = np.unique(labels)[1: ]
+    mean_int = np.zeros((images.shape[0], index.shape[0]))
+
+    for n in range(images.shape[0]):
+        mean_int[n] = meas.mean(images[n], labels, index=index)
+
+    return mean_int
+
+
+def combine_mean_intensity(mean_int_dict):
+    """
+    Parameters
+    ----------
+    mean_int_dict : dict
+        mean intensity of each ROI as a dictionary
+        {roi 1: average intensities, roi 2 : average intensities}
+
+    Returns
+    -------
+    combine_mean_int : array
+        combine mean intensities of image sets for each ROI of labeled array
+
+    """
+    return np.vstack(list(mean_int_dict.values()))
+
+
+def circular_average(image, calibrated_center, threshold=0, nx=100,
+                     pixel_size=None):
+    """
+    Circular average(radial integration) of the intensity distribution of
+    the image data.
+
+    Parameters
+    ----------
+    image : array
+        input image
+
+    calibrated_center : tuple
+        The center in pixels-units (row, col)
+
+    threshold : int, optional
+        threshold value to mask
+
+    nx : int, optional
+        number of bins
+
+    pixel_size : tuple, optional
+        The size of a pixel in real units. (height, width). (mm)
+
+    Returns
+    -------
+    bin_centers : array
+        bin centers from bin edges
+        shape [nx]
+
+    ring_averages : array
+        circular integration of intensity
+    """
+    radial_val = core.radial_grid(calibrated_center, image.shape,
+                                  pixel_size)
+
+    bin_edges, sums, counts = core.bin_1D(np.ravel(radial_val),
+                                          np.ravel(image), nx)
+    th_mask = counts > threshold
+    ring_averages = sums[th_mask] / counts[th_mask]
+
+    bin_centers = core.bin_edges_to_centers(bin_edges)
+
+    return bin_centers, ring_averages
```

Comment:
That is very contrary to expectations.  If they are not the same length how do you till with averages go with which bin?